### PR TITLE
Add example for sync gauge in Go

### PR DESCRIPTION
**Pull request by bagmeg**
Added example for sync gauge in Go.
This is part of following issue:
open-telemetry/opentelemetry-go#5414

### CODE REVIEW

File: content/en/docs/languages/go/instrumentation.md [hunk 1]
```diff
@@ -534,6 +534,34 @@ func removeItem() {
 }
 ```
 
+### Using Gauges
+
+Gauges are used to measure non-additive values when changes occur.
+
+For example, here's how you report the speed of a CPU fan:
```
Comment (cartermp) [inline content/en/docs/languages/go/instrumentation.md:541]:
```suggestion
For example, here's how you might report the current speed of a CPU fan:
```

File: content/en/docs/languages/go/instrumentation.md [hunk 2]
```diff
@@ -534,6 +534,34 @@ func removeItem() {
...
+	speedGauge, err := meter.Int64Gauge(
+		"cpu.fan.speed",
+		metric.WithDescription("Speed of CPU fan"),
+		metric.WithUnit("RPM"),
+	)
+	if err != nil {
+		panic(err)
+	}
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		speedGauge.Record(r.Context(), 1500)
```
Comment (cartermp) [inline content/en/docs/languages/go/instrumentation.md:560]:
```suggestion
		// hard-code 1500 RPM for demonstrative purposes
		speedGauge.Record(r.Context(), 1500)
```

**Review by cartermp — APPROVED** [review]
Overall LGTM, just some minor tweaks

File: content/en/docs/languages/go/instrumentation.md (hunk 2 above)
Comment (bagmeg) [reply content/en/docs/languages/go/instrumentation.md:561]: Thanks for the suggestion. Applied and pushed.

**Review by dmathieu — APPROVED** [review]

File: content/en/docs/languages/go/instrumentation.md [hunk 3]
```diff
@@ -534,6 +534,35 @@ func removeItem() {
 }
 ```
 
+### Using Gauges
```
Comment (theletterf) [inline content/en/docs/languages/go/instrumentation.md:537]:
```suggestion
### Using gauges
```

We use sentence case in titles.

**Review by theletterf — APPROVED** [review]
Just a tiny nit. Thanks!

File: content/en/docs/languages/go/instrumentation.md (hunk 3 above)
Comment (bagmeg) [reply content/en/docs/languages/go/instrumentation.md:537]: Thank you for the feedback. I noticed that other titles like 'Using Histograms' also start with capital letters.  Could you give me more detail on which part  i should fix?

**Pull request merged**